core/src/main/java/org/commonjava/indy/core/content/DefaultDownloadManager.java

```diff
@@ -300,16 +306,23 @@ public DefaultDownloadManager( final StoreDataManager storeManager, final Transf
 
     @Override
     @Measure
-    public List<StoreResource> list( final List<? extends ArtifactStore> stores, final String path, final EventMetadata eventMetadata )
+    public List<StoreResource> list( final List<? extends ArtifactStore> stores, final String path,
+                                     final EventMetadata eventMetadata )
             throws IndyWorkflowException
     {
         final String dir = PathUtils.dirname( path );
 
         final List<StoreResource> result = new ArrayList<>();
         try
         {
+            List<? extends KeyedLocation> locs = LocationUtils.toLocations( stores );
+            for ( KeyedLocation l : locs )
+            {
+                l.setAttribute( Location.AS_DIRECTORY, Boolean.TRUE );
```

Comment:
Is there a way to use the event metadata to pass the flag? KeyedLocation sounds like a read-only thing. We always use the eventmeta for such porpose. Conceptually we may not want to use another way passing though per-req params.

Reply:
Unfortunately I did not find a very good way to do this without less api changes than this one. From what I understand, I think this flag can be accepted because it used to identify the location's attribute, and this "AS_DIRECTORY" can be considered as sort of the attribute of the location to identify this location should be considered as a directory but not others.

Reply:
This sounds good argument. LGTM.